src/components/NewDatePicker/index.js

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import {View, Animated} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CalendarPicker from '../CalendarPicker';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes as datePickerPropTypes, defaultProps as defaultDatePickerProps} from './datePickerPropTypes';
+import './styles.css';
+
+const propTypes = {
+    ...datePickerPropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultDatePickerProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedDate: new Date(),
+            isPickerVisible: false,
+        };
+
+        this.setDate = this.setDate.bind(this);
+        this.showPicker = this.showPicker.bind(this);
+        this.hidePicker = this.hidePicker.bind(this);
+
+        this.opacity = new Animated.Value(0);
+
+        /* We're using uncontrolled input otherwise it wont be possible to
+        * raise change events with a date value - each change will produce a date
+        * and make us reset the text input */
+        this.defaultValue = props.defaultValue
+            ? moment(props.defaultValue).format(CONST.DATE.MOMENT_FORMAT_STRING)
+            : '';
+    }
+
+    /**
+     * Trigger the `onInputChange` handler when the user input has a complete date or is cleared
+     * @param {Date} selectedDate
+     */
+    setDate(selectedDate) {
+        this.setState({selectedDate}, () => {
+            this.props.onInputChange(moment(selectedDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
+            this.hidePicker();
+        });
+    }
+
+    /**
+     * Function to show up and animates the picker.
+     */
+    showPicker() {
+        this.setState({isPickerVisible: true}, () => {
+            Animated.timing(this.opacity, {
+                toValue: 1,
+                duration: 100,
+                useNativeDriver: true,
+            }).start();
+        });
+    }
+
+    /**
+     * Function to animate and hide the picker.
+     */
+    hidePicker() {
+        Animated.timing(this.opacity, {
+            toValue: 0,
+            duration: 100,
+            useNativeDriver: true,
+        }).start((animationResult) => {
+            if (!animationResult.finished) {
+                return;
+            }
+            this.setState({isPickerVisible: false});
+            this.textInputRef.blur();
```

Comment:
Do we need to blur the input after hiding the picker?

Reply:
I see no harm in keeping it focused. I believe that's how it works on html5 date picker. But if it's required to be blurred then yeah `setDate` looks like a better place.